### ISSUE TITLE
Window borders don't seem to work

### ISSUE THREAD

**Issue by justbuchanan** [eXeC64/imv#6]
Hi, I saw your post on reddit and decided to check this out!  I like it pretty well, although one minor thing I noticed is that it doesn't draw any window borders.  I have my tiling wm (bspwm) setup so that the focused window draws a small border.  How hard would this be to implement for `imv`?

**Comment by eXeC64** [comment]
That's odd. There's no reason there shouldn't be window borders, and they show up for me in i3. Do you have a sample bspwm config I could use that reproduces the problem?

**Comment by justbuchanan** [comment]
Sure thing, here's my [bspwmrc](https://github.com/justbuchanan/dotfiles/blob/master/roles/arch/files/home/.config/bspwm/bspwmrc).

**Comment by autrimpo** [comment]
I use bspwm as well, so I tried xprop to see if there's anything different from the other windows - this is what I found, which I guess bspwm uses instead of its own borders.
`_MOTIF_WM_HINTS(_MOTIF_WM_HINTS) = 0x2, 0x0, 0x1, 0x0, 0x0, 0x8e5b6f30, 0xff, 0x8e5b6f30, 0x3c0000d, 0x8e6bd7d0`

I made a quick search through the code and found nothing, so I guess it's something SDL does on its own?

**Comment by eXeC64** [comment]
I've had a look at some documentation and the third value in the hints array is for decorations (0x1) so they ought to be enabled.

@autrimpo Since you already run bspwm, could you try adding a `SDL_SetWindowBordered(window, 1);` call to main somewhere after creating the window? Maybe bspwm just needs a little help. If that works please check it also works when going in/out of fullscreen. That function may be needed there too.

**Comment by autrimpo** [comment]
Sure, will do that.

**Comment by eXeC64** [comment]
Thanks.

**Comment by autrimpo** [comment]
Unfortunatelly that doesn't work. I've downloaded the SDL source and the `SetWindowBordered` function gets called when creating a window anyway. It is the only function that does something with the `_MOTIF_WM_HINTS` atom. I want to try to remove the atom using Xlib, but that feels a bit hacky.

**Comment by eXeC64** [comment]
>  I want to try to remove the atom using Xlib, but that feels a bit hacky.

Yeah. It'd also mean adding a direct dependency on X11, which I'd very much like to avoid. We should preferably be able to do everything display/wm specific via SDL2.

This is sounding like a bug with SDL2 itself, so it's probably best if we figure out what's going on and submit a bug report upstream.

How much of a problem is this bug visually? Is it a minor annoyance/inconsistency, or a big pain for bspwm users? If it's not critical and we can't easily work around it ourselves I'd like to move this back from the v1.1.0 release and get it fixed upstream.

**Comment by autrimpo** [comment]
Visually I'd say pretty minor, though I have only 4px borders, for bigger borders I can imagine it looks way worse. 

What's worse is the lack of feedback about the state of the window, be it simply active/inactive or other states that bspwm has (protected, sticky...). Honestly I didn't even notice the bug until I saw this report, but I guess that depends a lot on the specific setup and workflow.

**Comment by eXeC64** [comment]
I think this can be put on the back burner for now then. Definitely need to try and get it fixed upstream though.

**Comment by autrimpo** [comment]
I tried some more things today.
Calling the `SetWindowBordered` function in various places, adding the `SDL_WINDOW_BORDERLESS` flag to the window when creating it.
Not only did it not affect the bug, the values of `_MOTIF_WM_HINTS` did not change either.

So I built a patched version of SDL that had the whole `SetWindowBordered` function commented out.
As expected, the imv window no longer had the `_MOTIF_WM_HINTS` atom.
However, the mysterious black border still remains.

Maybe this isn't a SDL bug after all, but a bspwm bug? Although I have no idea how or why, because the xprop output shows nothing suspicious.

I guess I will create a bspwm issue and see what response I get there.
